src/Features/CSharp/Portable/UsePatternMatching/CSharpIsAndCastCheckWithoutNameDiagnosticAnalyzer.cs

```diff
@@ -54,9 +53,17 @@ public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
             => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
 
         protected override void InitializeWorker(AnalysisContext context)
-            => context.RegisterSyntaxNodeAction(SyntaxNodeAction, SyntaxKind.IsExpression);
+        {
+            context.RegisterCompilationStartAction(context =>
+            {
+                var expressionTypeOpt = context.Compilation.GetTypeByMetadataName("System.Linq.Expressions.Expression`1");
+                context.RegisterSyntaxNodeAction(context => SyntaxNodeAction(context, expressionTypeOpt), SyntaxKind.IsExpression);
```

Comment:
Is this the only analyzer that needs to handle this case? For example:

https://github.com/dotnet/roslyn/blob/main/src/Analyzers/CSharp/Analyzers/UseIsNullCheck/CSharpUseIsNullCheckForReferenceEqualsDiagnosticAnalyzer.cs

Reply:
@CyrusNajmabadi Is this being tracked somehow? Or is no change needed at all?

Reply:
in general we are ok with a "wait and see" approach.  lots of issues/work to do, and night infinite scenarios that might exist out there.  if it's not negatively impacting people, better to spend time on things that are an issue vs speculating.